Implemented color theme functionality and added tests

opobrien opened the pull request:
Description:
- Added color_theme management to the Instructor class to set and get themes (light/dark).
- Implemented display_courses() method to show courses an instructor is teaching.
- Added unit tests for theme management and course display.
- All tests passed successfully.

Issue:#67

sjiang1 commented on instructor.py:
```diff
@@ -1,8 +1,8 @@
 from data import INSTRUCTORS, COURSES
-from color_ui import print_information
 
 class Instructor:
     def __init__(self, instructor_id):
+        self.color_theme = "light"  # Default theme is light
         if instructor_id in INSTRUCTORS:
             self.instructor_id = instructor_id
             self.name = INSTRUCTORS[instructor_id]
```
I don't see a place where you use the color_theme? The prompt doesn't change color based on the color_theme.

sjiang1 reviewed (changes requested):
Please also make sure your branch is up to date with main.

sjiang1 commented:
Please also add your issue id here.

opobrien commented:
All updates have been completed. I integrated the theme selection functionality into main.py using apply_theme(), and updated color_ui.py to support dynamic light and dark themes. I also removed the outdated theme logic from the Instructor class

sjiang1 commented:
I haven't checked your code, but the tests didn't pass.

opobrien commented:
Yeah I had not finished fixing all the tests but it should be okay now.

On Fri, Apr 11, 2025 at 7:41 PM Siyuan Jiang ***@***.***>
wrote:

> I haven't checked your code, but the tests didn't pass.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/COSC381-2025Winter/GradebookSystem/pull/86#issuecomment-2798221848>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BO5ZFMH4XPVKJCMUAQGCA7D2ZBHMNAVCNFSM6AAAAAB24QS4GCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOOJYGIZDCOBUHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> *sjiang1* left a comment (COSC381-2025Winter/GradebookSystem#86)
> <https://github.com/COSC381-2025Winter/GradebookSystem/pull/86#issuecomment-2798221848>
>
> I haven't checked your code, but the tests didn't pass.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/COSC381-2025Winter/GradebookSystem/pull/86#issuecomment-2798221848>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BO5ZFMH4XPVKJCMUAQGCA7D2ZBHMNAVCNFSM6AAAAAB24QS4GCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOOJYGIZDCOBUHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

sjiang1 commented on color_theme.py:
```diff
@@ -0,0 +1,53 @@
+    Apply the color theme globally using set_theme_colors.
+    """
+    if theme == "light":
+        set_theme_colors({
+            "success": Fore.GREEN,
+            "error": Fore.RED,
+            "warning": Fore.YELLOW,
+            "info": Fore.BLUE
+        })
+    elif theme == "dark":
+        set_theme_colors({
+            "success": Fore.LIGHTGREEN_EX,
+            "error": Fore.LIGHTRED_EX,
+            "warning": Fore.LIGHTYELLOW_EX,
+            "info": Fore.CYAN
+        })
```
Great work! It's very hard to tell the difference in the colors of the light and dark themes. In the dark mode, please add a background color for the dark theme, for example, try Back.black with other foreground color. Thanks!

sjiang1 reviewed (changes requested):
Very well done! Please add the last bit of change, and you are almost there 👍

opobrien commented:
I believe this is now ready for review

teststudentcosc381 reviewed (approved):
Good job!
